### DISCUSSION TITLE
Using keystone with bun

### DISCUSSION THREAD

**Discussion by vkuksa** [keystonejs/keystone#9091] [Questions]
Is there a proper way of integrating them together? Or keystone is completely node-bound and there's no reason behind this experiment?
Maybe there are plans over making such port?

**Comment by unrevised6419** [comment]
> At its core is the Bun runtime, a fast JavaScript runtime designed as a drop-in replacement for Node.js.
> – https://bun.sh/docs

It should work out of the box. What exactly did you try that did not work? 🤔

**Reply by unrevised6419** [reply]
It just works.

<img width="1663" alt="image" src="https://github.com/keystonejs/keystone/assets/1881266/ed3abbfb-80d0-4a92-a1de-9dd59770ffba">

```shell
bun install
bun install v1.1.7 (b0b7db5c)

$ keystone build --no-ui --frozen
✨ GraphQL and Prisma schemas are up to date

Checked 926 installs across 910 packages (no changes) [1108.00ms]
bun run dev
$ keystone dev
✨ Starting Keystone
⭐️ Server listening on :3000 (http://localhost:3000/)
⭐️ GraphQL API available at /api/graphql
✨ Generating GraphQL and Prisma schemas
✨ The database is already in sync with the Prisma schema
✨ Connecting to the database
✨ Creating server
✅ GraphQL API ready
✨ Generating Admin UI code
✨ Preparing Admin UI app
✅ Admin UI ready
^C
bun run build
$ keystone build
✨ Generated GraphQL and Prisma schemas
✨ Generating Admin UI code
✨ Building Admin UI
   Skipping validation of types
   Skipping linting
 ✓ Creating an optimized production build
 ✓ Compiled successfully
 ✓ Collecting page data
 ✓ Generating static pages (15/15)
 ✓ Collecting build traces
 ✓ Finalizing page optimization

Route (pages)                              Size     First Load JS
┌ ○ /                                      1.66 kB         507 kB
├   /_app                                  0 B             506 kB
├ ○ /404                                   178 B           506 kB
├ ○ /init                                  6.64 kB         512 kB
├ ○ /no-access                             414 B           506 kB
├ ○ /posts                                 240 B           515 kB
├ ○ /posts/[id]                            244 B           510 kB
├ ○ /posts/create                          1.5 kB          507 kB
├ ○ /signin                                1.59 kB         507 kB
├ ○ /tags                                  242 B           515 kB
├ ○ /tags/[id]                             245 B           510 kB
├ ○ /tags/create                           1.5 kB          507 kB
├ ○ /users                                 239 B           515 kB
├ ○ /users/[id]                            243 B           510 kB
└ ○ /users/create                          1.5 kB          507 kB
+ First Load JS shared by all              506 kB
  ├ chunks/framework-8a8ee0a30377ce1f.js   45.4 kB
  ├ chunks/main-6ca10d9827ab70e7.js        32.9 kB
  ├ chunks/pages/_app-925bc9eda3fc7b59.js  426 kB
  └ chunks/webpack-9d79e7a696dc200a.js     1.03 kB

○  (Static)  automatically rendered as static HTML (uses no initial props)

bun run start
$ keystone start
✨ Starting Keystone
✨ Connecting to the database
✨ Creating server
✅ GraphQL API ready
✨ Preparing Admin UI Next.js app
✅ Admin UI ready
⭐️ Server listening on :3000 (http://localhost:3000/)
```

**Reply by vkuksa** [reply]
Hello, Andrei! Thank you for your replies and displaying interest in this question.

I had existing cms service written in node-driven keystone. It was using sqlite as a storage. 

For a sake of integration with bun, i tried:
- copying repo and installing dependencies with `bun install`
- building and running service with `bun --bun run build` + `bun --bun run start` (using --bun flag as described in https://bun.sh/docs/cli/run#bun) (without --bun flag everything was working as with node) 
- using dev target with `bun --bun run dev`

Service was running successfully when starting `dev` target, but did not start when using `start` target, as (i suspect, so further writing is speculation), it was not connecting properly to database with prisma. Also i tried using postgres and got similar results.

**Reply by unrevised6419** [reply]
> as (i suspect, so further writing is speculation), it was not connecting properly to database with prisma. Also i tried using postgres and got similar results.

You are not in the wrong. It stops exactly at [this line](https://github.com/keystonejs/keystone/blob/623592bfb9139d8fb2a162a997d187be73daaa61/packages/core/src/lib/createSystem.ts#L246)

It seems Prisma has a lot of issues with topic bun https://github.com/prisma/prisma/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+bun+label%3A%22topic%3A+bun%22

<img width="465" alt="image" src="https://github.com/keystonejs/keystone/assets/1881266/5981671f-d38d-4515-ab9f-223c47b87568">

<img width="879" alt="image" src="https://github.com/keystonejs/keystone/assets/1881266/ae819d2c-ced2-4abe-b3d6-8bd71269c865">
